packages/atoms/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from "react";
+
+import { classNames } from "@calcom/lib";
+
+import { Button } from "../src/components/ui/button";
+import type { AtomsGlobalConfigProps } from "../types";
+import type { ConnectButtonProps } from "../types";
+
+// we accept two props: buttonText & onButtonClick
+// buttonText is for displaying buttonText
+// onButtonClick is a a method to handle onClick for button
+// it accepts another object which contains a callback in case of any error
+
+export function ConnectButton({
+  buttonText,
+  onButtonClick,
+  stylesClassname,
+  icon,
+}: ConnectButtonProps & AtomsGlobalConfigProps) {
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+  const [errMsg, setErrMsg] = useState<string>("");
+
+  function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+
+    setIsProcessing(true);
+    onButtonClick({
+      cb: (err) => {
```

Comment:
I think this approach is too complicated, 

we should aim to provide simple interface to the atoms and one that people are used to, meaning props such as onChange, onSuccess, onError, onClick...